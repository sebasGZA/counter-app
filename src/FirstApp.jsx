import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{title}</h1>

      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  subTitle: PropTypes.string,
  title: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  name: "Sebastian Gomez",
  subTitle: "Anything",
  title: "No title",
};
