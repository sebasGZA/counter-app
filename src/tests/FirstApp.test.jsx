import { render } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe("Tests in <FirstApp/>", () => {
  const title = "Hello, I'm Sebastian";

  test("should make match with snapshot", () => {
    const { container } = render(<FirstApp title={title} />);

    expect(container).toMatchSnapshot();
  });

  test("should show title into a h1 tag", () => {
    const { container, getByText } = render(<FirstApp title={title} />);
    expect(getByText(title)).toBeTruthy();

    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
