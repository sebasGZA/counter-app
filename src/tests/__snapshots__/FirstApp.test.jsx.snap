// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp/> should make match with snapshot 1`] = `
<div>
  <h1>
    Hello, I'm Sebastian
  </h1>
  <p>
    123
  </p>
  <p>
    Sebastian Gomez
  </p>
</div>
`;
